net.bobbo.player-controller/player_scripts/crouch.gd

```diff
@@ -33,4 +40,37 @@ func player_physics_process(delta: float):
 	var scale_difference = new_scale - previous_scale
 	
 	player.velocity.y += scale_difference * 0.5
-	player.height = new_scale
+	
+	# Make it so that the crouch can only go as low as regularly crouching and only high enough such that the player doesn't get stuck if an object is detected above them
```

Comment:
With this approach, the player can sort of "half-crouch" by being in spaces shorter than the standing height, but larger than the crouch height.

Instead of a sort of sliding scale, can you revise this to act more boolean? You should be able to get away with just setting the crouch target height to 0.5 if it detects a potential ceiling collision.

Reply:
I can totally do that and now thinking about it I don't know why I thought it would be better to go through the trouble of making equations that would calculate the "sliding scale", it totally would have been easier to do that and is much less code. Definitely updating that